pydra/engine/core.py

```diff
@@ -924,17 +1070,24 @@ def __init__(
 
         self.graph = DiGraph(name=name)
         self.name2obj = {}
+        self._lzin = None
+        self._pre_split = (
+            False  # To signify if the workflow has been split on task load or not
+        )
 
         # store output connections
         self._connections = None
         # propagating rerun if task_rerun=True
         self.propagate_rerun = propagate_rerun
 
+    @property
```

Comment:
I went to do this and realised that despite being stored the property isn't actually being cached in the current code. Not sure whether it should be or not (I'm thinking not). Checking that the checksum of a node doesn't change during execution is something I think we catch explicitly otherwise Pydra can't find the results

Reply:
I am confused whether your comment means `lzin` could become a `cached_property` or not.

Reply:
ah yes, in my VS code plugin your comment was showing up on a different property. I have changd it now in my branch